ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py

```diff
@@ -316,14 +316,24 @@ def _wan_load_balance_policies(self) -> list:
                     context_keys=["name"],
                 )
 
-            default_virtual_topology = get(policy, "default_virtual_topology", required=True)
+            default_virtual_topology = get(
```

Comment:
Something seems wrong about the order here. You error out suggesting the drop_unmatched _above_ the line where you check for drop_unmatched.

Reply:
so not really - line 319 is checking for the presence of the key that must be there
the second line is checking that either `drop_unmatched` is set (and then we render nothing) or then we render path-groups.

This is to force explicit configuration of `defaut_virtual_topology` every time as we discussed - if you want to drop unmatched then you need to do it explicitly rather than having it done silently for you (I need to check if the dict is required in schema)

Reply:
The error message on line 319 is that the default_virtual_topology must be specified. This will error out no matter what since you have required=True. So setting `drop_unmatched` to True will not stop that.

Reply:
yes thats the goal
1. you must define `default_virtual_topology` with either `drop_unmatched` or something else - line 319 checks that it is defined if not
2. either you have `drop_unmatched` and nothing is rendered or we render the rest

Reply:
For someone getting this error, this last line will only confuse them. I think this org_key should just be the full variable path to the missing key. But it would be simpler to just require it in the schema and just use `['default_virtual_topology']` here.